### ISSUE TITLE
SIGSEGV in dynamic when key not in context

### ISSUE THREAD

**Issue by tonsky** [HumbleUI/HumbleUI#31]
Ok, good to know, also if you accidentally require a key in ui/dynamic which is not in context the jvm dies with a `SIGSEGV (0xb) at pc=0x00000001526f15f9, pid=46668, tid=775`.

I've been assuming that you've been doing this for perf reasons, but I thought I'd just check because at times it's non-obvious which component is missing what.

_Originally posted by @Folcon in https://github.com/HumbleUI/HumbleUI/issues/13#issuecomment-1128190260_

**Comment by tonsky** [comment]
@Folcon what do you mean? Like this

```
(ui/dynamic ctx [key (:unknown ctx)]
```

This works fine for me

**Comment by Folcon** [comment]
Ok looking at my crashdumps, I think I've made a mistake, the issue occurs, but the cause is not `ui/dynamic` misbehaving specifically.

The issue is components which are created which when I use them in another context and they're missing expected keys causes a jvm crash, as libskija is less forgiving.

For example:
```java
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00000001506f15f9, pid=51839, tid=775
#
# JRE version: OpenJDK Runtime Environment Homebrew (17.0.2) (build 17.0.2+0)
# Java VM: OpenJDK 64-Bit Server VM Homebrew (17.0.2+0, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-amd64)
# Problematic frame:
# C  [libskija_x64.dylib+0x2f15f9]  SkPaint::nothingToDraw() const+0x9
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

~~~ ELIDED ~~~

---------------  T H R E A D  ---------------

Current thread (0x00007fe41200bc00):  JavaThread "main" [_thread_in_native, id=775, stack(0x00007ffee2515000,0x00007ffee6514000)]

Stack: [0x00007ffee2515000,0x00007ffee6514000],  sp=0x00007ffee65099a0,  free space=65490k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libskija_x64.dylib+0x2f15f9]  SkPaint::nothingToDraw() const+0x9
C  [libskija_x64.dylib+0x264f57]  SkCanvas::internalQuickReject(SkRect const&, SkPaint const&, SkMatrix const*)+0x47
C  [libskija_x64.dylib+0x266c0f]  SkCanvas::onDrawRect(SkRect const&, SkPaint const&)+0x1f
C  [libskija_x64.dylib+0x265605]  SkCanvas::drawRect(SkRect const&, SkPaint const&)+0xb5
C  [libskija_x64.dylib+0x7d9b]  Java_io_github_humbleui_skija_Canvas__1nDrawRect+0x2b
J 7455  io.github.humbleui.skija.Canvas._nDrawRect(JFFFFJ)V (0 bytes) @ 0x00000001229120a9 [0x0000000122911fa0+0x0000000000000109]
j  io.github.humbleui.skija.Canvas.drawRect(Lio/github/humbleui/types/Rect;Lio/github/humbleui/skija/Paint;)Lio/github/humbleui/skija/Canvas;+67
```

The problem may not be solvable, but yea, getting a better idea of what's wrong while developing and preferably not having crashes would be good.

If that's out of scope, sorry, feel free to close =)...

**Comment by tonsky** [comment]
Actually... Try enabling asserts? `-ea` is JVM option. It should check for nulls in Skija and throw instead of dying silently

**Comment by Folcon** [comment]
So good news is that it's stopped crashing, bad news if the assert exception propagates through it seems to lockup everything.

The repl is active, but each time I try to reload a good state, it keeps throwing the same exception.

Hmm, I'll see if I can catch the exception better.

**Comment by tonsky** [comment]
Should’ve been fixed in acead0cfec02660e2ce8885bc40d23fdd118bd90

**Comment by Folcon** [comment]
@tonsky, you might want to catch `Throwable` instead of `Exception` as asserts cause similar "locking up" behaviour?

Or is that too much?

**Comment by tonsky** [comment]
Thanks @Folcon, good idea
